realm/realm-library/src/main/java/io/realm/internal/TableOrView.java

```diff
@@ -372,10 +382,13 @@
     Table pivot(long stringCol, long intCol, PivotType pivotType);
 
     /**
-     * Syncs the tableview with the underlying table data. It is not required to call this explicitly, all other API
-     * methods will automatically sync the view as well.
+     * Syncs the tableview with the underlying table data. This is effectively the same as rerunning the query, so it
+     * should not be called on TableViews created by an async query.
+     *
+     * This method gets automatically called when calling {@link Realm#refresh()} or when another threads updates
+     * the Realm, but it will _not_ be called the same thread commits a transaction.
      *
```

Comment:
`another threads` ->  `another thread`
`called the same thread commits` -> `called on the same thread committing`